src/type-layout.md

```diff
@@ -73,6 +73,12 @@ at least equal to the size and alignment of a pointer.
 > title="Dynamically Sized Types">DSTs</abbr> are currently twice the size of
 > the size of `usize` and have the same alignment.
 
+## `str` Layout
+
+Strings are laid out so that the `nth` character of the string is offset from the
```

Comment:
We call these "string slices" in the standard library documentation to differentiate from the owned `String` type.